### ISSUE TITLE
The requested URL returned error: 404

### ISSUE THREAD

**Issue by IvanKravchenko773** [convictional/trigger-workflow-and-wait#62]
Hi, I tying to trigger workflow but not in the main branch and it wont work. I used Argument (ref:) with name of branch but always get 404 error. Am I doing something wrong? 
![image](https://user-images.githubusercontent.com/89259684/192969070-56a05fef-372d-4a1d-8f41-6f5b864ae1ae.png)

**Comment by rabih** [comment]
The only thing that comes to mind is the private key you are connecting with does not have access to that repository or that the workflow path is incorrect. "workflows/test3.yml/dispatches" doesn't seem like a proper path.

**Comment by tommoyang** [comment]
I was running into this error myself despite using the same setup in other repositories fine - turns out that it's expecting that a workflow exists _before_ the API calls will start working. I manually triggered a workflow run, and then this action started working

**Comment by AshwinNmbr** [comment]
@IvanKravchenko773 Did you find a solution? I am facing the same exact error too :(

**Comment by IvanKravchenko773** [comment]
@AshwinNmbr no, but I think it just cant work if yml file is not in main branch.

**Comment by AshwinNmbr** [comment]
I managed to fix it by giving token in the args

**Comment by Davide-DD** [comment]
Hi everyone, I'm having the exact same problem: I'm trying to execute a workflow in an organization repo (where I'm the owner) but I cannot make it work. I'm specifying both the token and the workflow file name (and also owner and repository path like organization_name/repo_name).
@AshwinNmbr what do you mean by "giving PAT in the args"? The github_token?

**Comment by Davide-DD** [comment]
In my case I had a repo with master branch and the other with main branch, so I had to specify "ref" in order to make it work

**Comment by scottdickerson** [comment]
does anyone know which specific permissions are needed by the PAT?  Is it possible to grant these permissions to the OOTB GITHUB_TOKEN?

**Comment by stormeye2000** [comment]
I had the same problem, api failed 404
So i made sure that the demo worked [here](https://keithweaver.ca/lessons/trigger-another-repositorys-github-action-workflow-wait-for-result)
And then I added my specific action jobs to the called action
It looks like the sleep commands solved it
`    steps:
      - name: Setting Up Integration Tests
        run: |
          echo "Wait for NCTL"
          sleep 10s
`

**Comment by anupkumardh** [comment]
Had the same issue was forgot to add the user to the repo

**Comment by littlix** [comment]
For people who are as dumb as I - if you're doing this in a repo you've created in an organisation, the owner is the organisation, not yourself 💀

**Comment by radnov** [comment]
I encountered this error today and after a few hours of scratching my head I realised I was setting an environment variable named `API_URL` in a previous job step, which was picked up by this action [here](https://github.com/convictional/trigger-workflow-and-wait/blob/f69fa9eedd3c62a599220f4d5745230e237904be/entrypoint.sh#L14). 🤦 

Not that this is a bug, but using such general environment variable names with parameter expansion in public facing scripts is bound to lead to mishaps like this.

**Comment by brianjmurrell** [comment]
> does anyone know which specific permissions are needed by the PAT? Is it possible to grant these permissions to the OOTB GITHUB_TOKEN?

Did you find an answer to either (or both!) of these questions?  The README is light (read: absent) of any details about the scopes needed on the PAT.

**Comment by danopia** [comment]
I'm looking into the fine-grained permissions for this action.. The APIs that I see used with the `github_token` input, and their permissions, are:

* `repos/${INPUT_OWNER}/${INPUT_REPO}/actions/workflows/${INPUT_WORKFLOW_FILE_NAME}/runs`
  * Needs "Actions" repository permissions (read)
* `repos/${INPUT_OWNER}/${INPUT_REPO}/actions/workflows/${INPUT_WORKFLOW_FILE_NAME}/dispatches`
  * Needs "Actions" repository permissions (write)
* `repos/${INPUT_OWNER}/${INPUT_REPO}/actions/runs/$last_workflow_id`
  * Needs "Actions" repository permissions (read)

In summary it looks like "Actions: Read and write" should be sufficient. I'll be validating this.

As far as the OOTB GITHUB_TOKEN, I understand that it is not possible to use because anything done using the Github Actions GITHUB_TOKEN is blocked from triggering further Github Actions.

**Comment by brianjmurrell** [comment]
Any relationship to #96 here?
